# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/ScalarType.h
# Commit: 5445f480023e
# Full Hash: 5445f480023e36e3dd1cea7bd659202ac83d25bc
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-09-09 15:22:41
# Regressor Bug: 1720422
# File Overlap Count: 2
# Description:
#   Bug 1720422 - Move js::Scalar -> JS::Scalar for public API, but alias it as js::Scalar r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D120292
# ==============================================================================

diff -r 988eccbeb2ae -r 5445f480023e js/public/ScalarType.h
--- a/js/public/ScalarType.h	Wed Sep 08 23:46:59 2021 +0000
+++ b/js/public/ScalarType.h	Wed Sep 08 23:47:00 2021 +0000
@@ -13,7 +13,7 @@
 
 #include <stddef.h>  // size_t
 
-namespace js {
+namespace JS {
 
 namespace Scalar {
 
@@ -217,6 +217,15 @@
 
 }  // namespace Scalar
 
+}  // namespace JS
+
+namespace js {
+
+// This is aliased in NamespaceImports.h, but that is internal-only and
+// inaccessible to Gecko code, which uses this type fairly heavily. Until such
+// uses are changed, we need the alias here as well.
+namespace Scalar = JS::Scalar;
+
 }  // namespace js
 
 #endif  // js_ScalarType_h